### DISCUSSION TITLE
Multi-tenancy implementation where users are shared among the organizations.

### DISCUSSION THREAD

**Discussion by ruslankiskinov** [zitadel/zitadel#8518] [Q&A]
Hi,
I want to get your opinion on how to implement a more unusual multi-tenant model. Here are the requirements:
I'll use the term organization in the broad meaning and will not have in mind the actual term used in the ZITADEL.
The overall model is similar to Facebook's users and groups/pages.

- There are two main entities:
     - Users - people who register on the platform. They will have only one login name.
     - Organizations - entities to which users can join. One person can be a member of many organizations without the need to create a user for every organization.
- Users do not belong to any "organization". The users can log into the platform and select an organization to join. 
- Users can register to the platform freely. The model is similar to the Facebook users and groups/pages
- Every organization has an administrator (like in the facebook groups/pages) who has to allow users to join the organization.
- Every organization has predefined roles that are the same in all organizations like admin, member, editor, etc.

My questions are:

- What would be the best approach to implement this with ZITADEL? Since the users are a common shared entity.
- What are the performance implications if I create thousands of organizations?
- What are the performance implications if I create thousands of projects inside the organization?

For example, can I create one organization that holds all users and then use projects to represent different organizations?
I'm trying to select the best solution for my project and I think ZITADEL is closer to what I need compared to other solutions I researched like KEYCLOAK, Authentik, and Ory.

Thank you in advance!

**Comment by hifabienne** [comment]
Hi @ruslankiskinov 

I would create a separat organization where you put all your users.

For the organizaitons to join i can see different possibilities.
In my opinion it doesn't directly relate to organizaitons as we know them in ZITADEL.

you could create a project with different roles with the organization names. and add a user to the project with the role of the organization.

Or you manage the organizations in your application and you just add the role like admin, member, editor, etc. to the project.

A different possiblity would also be to have a project per organization and then give the role admin, member, editor in there.

Just be aware, that out of the box a user can't just add itself a role. this would be an admin to do that, which you could implement with a machine/service account which has the right permission.

It feels like ZITADEL is not directly designed for your use case, but there are possiblities to solve it.

There is a feature which we do not yet hav, but this could also help you in the guture: https://github.com/zitadel/zitadel/issues/5822
